api/src/opentrons/protocol_api_experimental/labware.py

```diff
@@ -82,7 +81,7 @@ def parent(self) -> Union[DeckSlotName, Any]:
         )
         if isinstance(parent, DeckSlotLocation):
             return str(parent.slotName)
-        elif isinstance(parent, ModuleLocation):
+        else:
             raise NotImplementedError("Not yet implemented for labware on modules.")
```

Comment:
In this case, it's a little more type-safe to end with an `elif` instead of `else`, because it lets mypy enforce that we handle every case explicitly. (One of mypy's supported forms of [exhaustiveness checking](https://mypy.readthedocs.io/en/stable/literal_types.html#exhaustiveness-checking).)

Also, we should probably update this error message to account for off-deck labware.

So I would write this as:

```suggestion
        if isinstance(parent, DeckSlotLocation):
            return str(parent.slotName)
        elif isinstance(parent, ModuleLocation):
            raise NotImplementedError("Not yet implemented for labware on modules.")
        elif parent is None:
            raise NotImplementedError("Not yet implemented for off-deck labware.")
```

Edit: Oops, I just saw [your other comment](https://github.com/Opentrons/opentrons/pull/11521#discussion_r986003268) about this code being imminently deleted anyway. I would still do this `elif` tweak, but it's your call.